packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -287,7 +286,7 @@ function GridFilterForm(props: GridFilterFormProps) {
           onClick={handleDeleteFilter}
           size="small"
         >
-          <GridCloseIcon fontSize="small" />
```

Comment:
Removing the `fontSize="small"` prop modify the default appearance of the icon

## New icon
![image](https://user-images.githubusercontent.com/45398769/156341597-2e9fe386-7c4b-46a6-95ed-8633a62b64b6.png)
## Previouse icon
![image](https://user-images.githubusercontent.com/45398769/156341652-0b1a27f8-2ab3-4a2a-9c6e-0dc2d91f51a1.png)

Reply:
I've added a conditional logic to fix this issue but I believe that's not the best approach, since icons are used directly from svg's and there could be a component to render the icons, used globally with props such as size and etc.

Reply:
I have a suggested approach for using the icons more efficiently in my mind, but that'll require some time since they're being used in a lot of places. I'll create a separate PR for that once I'm able to figure out the best approact